### PR TITLE
[Release] Merge grep application project

### PR DESCRIPTION
**Pull request by Sai-Prateek**

### CODE REVIEW

File: core_java/grep/src/main/java/ca/jrvs/apps/grep/JavaGrepImpl.java [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+
+    JavaGrepImpl javaGrepImp = new JavaGrepImpl();
+    javaGrepImp.setRegex(args[0]);
+    javaGrepImp.setRootPath(args[1]);
+    javaGrepImp.setOutFile(args[2]);
+
+    try {
+      javaGrepImp.process();
+    } catch (Exception ex) {
+      javaGrepImp.logger.error(ex.getMessage(), ex);
```
Comment (jarviscanada) [inline core_java/grep/src/main/java/ca/jrvs/apps/grep/JavaGrepImpl.java]: javaGrepImp.logger.error("meaningful message", ex);

File: core_java/grep/src/main/java/ca/jrvs/apps/grep/JavaGrepLambdaImpl.java [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+    return filesList;
+  }
+
+  @Override
+  public List<String> readLines(File inputFile) {
+    List<String> lines = new ArrayList<>();
+    try (Stream<String> stream = Files.lines(Paths.get(inputFile.toURI()))) {
+      lines = stream.collect(Collectors.toList());
+    } catch (IOException e) {
+      logger.error(e.getMessage(), e);
```
Comment (jarviscanada) [inline core_java/grep/src/main/java/ca/jrvs/apps/grep/JavaGrepLambdaImpl.java]: logger.error("meaningful message", e);

File: core_java/grep/src/main/java/ca/jrvs/apps/practice/Helloworld.java [hunk 3]
```diff
@@ -0,0 +1,10 @@
+package ca.jrvs.apps.practice;
+
+class HelloWorld {
+
+  // Your program begins with a call to main().
+  // Prints "Hello, World" to the terminal window.
```
Comment (jarviscanada) [inline core_java/grep/src/main/java/ca/jrvs/apps/practice/Helloworld.java]: remove

**Review by jarviscanada — APPROVED** [review]
LGTM

**Comment by Sai-Prateek** [comment]
Made pull request changes

**Pull request closed without merging**